docs/content/en/contribute/software/dev-environ/_index.md

```diff
@@ -1,20 +1,79 @@
+You can view the video
+[here](https://www.youtube.com/watch?v=UcmULstMYXQ).
+
+To prepare to contribute to the Keptn project, we recommend that you:
+
+* Study the [Keptn documentation](https://lifecycle.keptn.sh/docs/)
+  to understand what Keptn does and how it works.
+* Familiarize yourself with the
+  [lifecycle-toolkit](https://github.com/keptn/lifecycle-toolkit)
+  repository, which is the primary repository for
+  Keptn software and documentation.
+  In particular, study the sections for the four main Keptn components:
+  
+  * [lifecycle-operator](https://github.com/keptn/lifecycle-toolkit/tree/main/lifecycle-operator)
+  * [metrics-operator](https://github.com/keptn/lifecycle-toolkit/tree/main/metrics-operator)
+  * [scheduler](https://github.com/keptn/lifecycle-toolkit/tree/main/scheduler)
+  * [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/tree/main/klt-cert-manager)
+
+  Each of these is described in the
+  [Architecture](../../../docs/architecture/)
+  section of the documentation
+  and most include a *README* file with more information.
+* Study the material in
+  [Technologies and concepts you should know](../../general/technologies).
+* Create an account for yourself on
+  [github.com](github.com)
+  if you do not already have an account.
+* Set up a private Github repository to use with your development.
+
+## View repository
+
+When you view the
+[lifecycle-toolkit](https://github.com/keptn/lifecycle-toolkit)
+repository, you see that Keptn is composed of multiple components,
+each of which is discussed in the Architecture
+[Architecture](../../../docs/architecture/)
+documentation:
+
+* Two Kubernetes operators, `metrics-operator` and `lifecycle-operatory`
+* Keptn `scheduler`
+* Keptn `cert-manager`
+
+At the top level of the repository,
+you also see the `runtimes` directory.
+This defines the runners that you can use when defining
+tasks to be run either pre- or post-deployment.
+These are discussed in
+[Runners and containers](../../../docs/implementing/tasks/#runners-and-containers).
+
+## Install software
+
 To test and develop software for the Keptn project,
 you need to install the following on your system:
 
 * [**Docker**](https://docs.docker.com/get-docker/): a tool for containerization,
-which allows software applications to run in isolated environments
-and makes it easier to deploy and manage them.
+  which allows software applications to run in isolated environments
+  and makes it easier to deploy and manage them.
 * A Kubernetes cluster running an appropriate version of Kubernetes.
   See [Supported Kubernetes versions](../../../docs/install/reqs.md/#supported-kubernetes-versions)
   for details.
-  If you need to set up a local Kubernetes cluster
-  we recommend Kubernetes-in-Docker(kind).
+  Most contributors create a local
+  Kubernetes-in-Docker(KinD) cluster.
+  This is adequate for developing software for Keptn.
+  See
+  [Kubernetes cluster](../../../../../docs/install/k8s/)
```

Comment:
this link is not working

Reply:
Actually it was, both locally and on netlify.  Whatever, I fixed it.